code/modules/mob/mob_holder.dm

```diff
@@ -102,6 +102,21 @@
 	name = "pAI"
 	desc = "It's a little robot."
 	icon_state = "pai"
+// I'm pretty sure this is obsolete, will test and remove later if so
+/obj/item/holder/pai/can_enter_storage(obj/item/storage/S, mob/user)
+	return FALSE
+
+/obj/item/holder/pai/attack_obj(obj/O, mob/living/user, params)
+	if(ismodcontrol(O))
+		var/created_drone = FALSE
+		var/obj/item/mod/control/control = O
+		for(var/obj/item/mod/module/drone/module in control.modules)
+			for(var/mob/living/mob in contents)
```

Comment:
you should check that the living mob in contents is a pai, so that if bullshit happens, random mobs dont get deleted